### DISCUSSION TITLE
XCom and DockerSwarmOperator not working

### DISCUSSION THREAD

**Discussion by mmiraglio** [apache/airflow#16028] [Q&A]
Hi everyone, I've been successfully using `DockerOperator` and `do_xcom_push=True` (to push the stdout to the next step) for a while. Since our Airflow installation will be running in Docker Swarm I would like to switch to `DockerSwarmOperator` but I couldn't manage to make `XCom` to work.

I don't see something like [this](https://github.com/apache/airflow/blob/335fa106e81ba8a446f5cda91d52527e282996be/airflow/providers/docker/operators/docker.py#L282).

```
                ret = None
                if self.do_xcom_push:
                    ret = (
                        self.cli.logs(container=self.container['Id'])
                        if self.xcom_all
                        else line.encode('utf-8')
                    )
                return ret
```

for the `DockerSwarmOperator` . Am I missing something here? 

Thanks in Advance

**Comment by lucas-av7** [comment]
I used DockerSwarm once and I couldn't do XCOM to work.

**Comment by alexsanderson-santos** [comment]
Yeah, I used `DockerSwarmOperator` with airflow `v1.10.12` and xcom not work.

**Comment by potiuk** [comment]
Yep. It does not seem to implement handling `do_xcom_push` or `xcom_all` - maybe somone here woudl like to contribute a fix ? It will be ablable in Airflow 2.1+ only when we release (as 1.10 is only there for critical fixes) but might be worth to add support for it. At the very least maybe someone could create an issue for that?

**Reply by mmiraglio** [reply]
Thanks @potiuk for your help. I omitted to say  I'm a Airflow 1.10 user. For Airflow 2 I guess it has been partially addressed. Logs are being pulled from the service but still no support for `do_xcom_push` nor `xcom_all`.

```
def _run_service(self) -> None:
...
...
...
        if self.enable_logging:
            self._stream_logs_to_output()
```

The point is that there is a bug opened when setting `enable_logging = True` where tasks keep Indefinitely running.  So that's a blocker to make `xcom` to work.

[TaskInstances do not succeed when using enable_logging=True option in DockerSwarmOperator](https://github.com/apache/airflow/issues/13675)
